src/test/kotlin/com/ampnet/userservice/controller/CoopControllerTest.kt

```diff
@@ -62,6 +69,46 @@ class CoopControllerTest : ControllerTestBase() {
         }
     }
 
+    @Test
+    fun mustGetErrorIfReCaptchaReturnsError() {
+        suppose("ReCAPTCHA verification failed") {
+            Mockito.`when`(reCaptchaService.validateResponseToken(testContext.reCaptchaToken))
+                .thenAnswer { throw ReCaptchaException("ReCAPTCHA verification failed") }
+        }
+
+        verify("Controller will return ReCaptcha error code") {
+            val request = CoopRequest("new-coop-a", "New Coop a", null, null, testContext.reCaptchaToken)
+            val result = mockMvc.perform(
+                post(coopPath)
+                    .content(objectMapper.writeValueAsString(request))
+                    .contentType(MediaType.APPLICATION_JSON)
+            )
+                .andExpect(status().isBadRequest)
+                .andReturn()
+            verifyResponseErrorCode(result, ErrorCode.REG_RECAPTCHA)
+        }
+    }
+
+    @Test
+    fun mustGetErrorIfReCaptchaReturnsLowScore() {
+        suppose("ReCAPTCHA verification failed") {
+            Mockito.`when`(reCaptchaService.validateResponseToken(testContext.reCaptchaToken))
+                .thenAnswer { throw ReCaptchaException("ReCAPTCHA score is too low") }
+        }
+
+        verify("Controller will return ReCaptcha error code") {
+            val request = CoopRequest("new-coop-a", "New Coop a", null, null, testContext.reCaptchaToken)
+            val result = mockMvc.perform(
+                post(coopPath)
+                    .content(objectMapper.writeValueAsString(request))
+                    .contentType(MediaType.APPLICATION_JSON)
+            )
+                .andExpect(status().isBadRequest)
+                .andReturn()
+            verifyResponseErrorCode(result, ErrorCode.REG_RECAPTCHA)
+        }
+    }
```

Comment:
This test is not needed. Covers the same case as the one above: `mustGetErrorIfReCaptchaReturnsError`. This is a specific error inside the mocked service which returns the same exception. Test specific errors in `ReCaptchaServiceTest`.